catalyst_voices_packages/catalyst_cardano_serialization/lib/src/rbac/x509_certificate.dart

```diff
@@ -0,0 +1,348 @@
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:asn1lib/asn1lib.dart';
+import 'package:catalyst_cardano_serialization/catalyst_cardano_serialization.dart';
+import 'package:equatable/equatable.dart';
+
+bool _registeredASN1Names = false;
+
+/// Before we can interact with the asn1lib we must register frequent names,
+/// however there's no single entry point to catalyst_cardano_serialization
+/// and we must expect different starting points. Therefore we call the method
+/// to register these names each time we need to interact with the asn1lib.
+void _registerASN1FrequentNames() {
```

Comment:
for fake of naming clarity. Can you split this into `_registerASN1FrequentNames` and `_ensureASN1FrequentNamesRegistered`?